### ISSUE TITLE
VSCode debugging for typescript

### ISSUE THREAD

**Issue by NickIliev** [NativeScript/nativescript-vscode-extension#51]
_From @ryc16 on June 23, 2016 19:53_

Need an advice for better debugging experience on VS Code. Here is my debugging experience.

1/ Currently, after a change made, I have to do "restart" debugging. Is there any way not to do "restart" after a change? Is there a livesync watch on debuging? If so, how to set it?
2/ I have never been able to do "attach to the device" even my emulator is running. So I have to do launch on device. I always got error "Connection timeout" from VSCode console. Any idea?

So it is not fun to debug as above both experience take long time. Any 

```
┌──────────────────┬─────────────────┬────────────────┬───────────────┐
│ Component        │ Current version │ Latest version │ Information   │
│ nativescript     │ 2.0.1           │ 2.0.1          │ Up to date    │
│ tns-core-modules │ 2.0.1           │ 2.0.1          │ Up to date    │
│ tns-android      │ 2.0.0           │ 2.0.0          │ Up to date    │

```

_Copied from original issue: NativeScript/NativeScript#2360_

**Comment by NickIliev** [comment]
Hi @ryc16 

The issue with **_Connection timeout_** is documented bug here](https://github.com/NativeScript/nativescript-vscode-extension/issues/46) and a fix is on the way with future release 0.3.

**Comment by ivanbuhov** [comment]
Hi @ryc16 

1. Currently, we have no support for livesyncing and debuging at the same time. The support for this should come in the CLI first, and then in the VS Code. It is in our roadmap.

2. "Attach on device" is used when trying to attach to physical device. For attaching to emulator, try "Attach to emulator".
